src/thunderbots/software/util/parameter/generate_parameters.py

```diff
@@ -0,0 +1,291 @@
+                        default=parameter["default"],
+                        quote="\"" if parameter["type"] in constants.QUOTE_TYPES else "",
+                        min_val=parameter["min"] if parameter["type"] in constants.RANGE_TYPES else None,
+                        max_val=parameter["max"] if parameter["type"] in constants.RANGE_TYPES else None
+                    ))
+
+            except KeyError as kerr:
+                print constants.AUTOGEN_FAILURE_MSG
+                sys.exit('Required key missing from config: {}, please check parameter {}'.format(
+                    kerr, key))
+
+        # if there is no current group, a new namespace must be created
+        elif group_name is None:
+            file_pointer.write(constants.CFG_NEW_NAMESPACE.format(
+                namespace=key
+            ))
+            __cfg_gen(param_info[key], file_pointer, group_name=key)
+
+        # if there already is a group, add to that namespace
+        else:
+            file_pointer.write(constants.CFG_SUB_NAMESPACE.format(
+                namespace=group_name,
+                sub_namespace=key
+            ))
+            __cfg_gen(param_info[key], file_pointer, group_name=key)
+
+#######################################################################
+#                            CPP Generator                            #
+#######################################################################
+
+
+def generate_header_cpp(param_info, output_path):
+    """Takes the input dictionary from the parsed yaml
+    and generates the respective cfg files needed
+
+    :param param_info: Return value of load_configuration, dict containing params
+    :param output_path: Where the cfgs should be saved to
+    :type param_info: dict
+    :type output_path: str
+
+    """
+    # open files
+    dynamic_parameters_h = open(output_path+constants.DYNAMIC_PARMETERS_HEADER, 'w')
+    dynamic_parameters_cpp = open(output_path+constants.DYNAMIC_PARMETERS_CPP, 'w')
+
+    # write the header to .h .cpp file
+    dynamic_parameters_h.write(constants.H_HEADER)
+    dynamic_parameters_cpp.write(constants.CPP_HEADER)
+
+    # generate cfg str used for topic subsription
+    cfg_strs = ",".join("\"{}\"".format(key) for key in param_info.iterkeys())
+    dynamic_parameters_cpp.write(constants.CFG_STR_VECTOR.format(cfg_strs))
+
+    # iterate through param_info starting from namespaces
+    # the first key is the file name, so that is skipped
+    for key, value in param_info.iteritems():
+        __header_cpp_gen(value, key, dynamic_parameters_h, dynamic_parameters_cpp)
+
+    # append the footer
+    dynamic_parameters_h.write(constants.FOOTER)
+    dynamic_parameters_cpp.write(constants.FOOTER)
+
+    # close files
+    dynamic_parameters_h.close()
+    dynamic_parameters_cpp.close()
+
+    print '===== created header ====='
+
+
+def __header_cpp_gen(param_info, cfg_name, header_file_pointer, cpp_file_pointer):
```

Comment:
similar to above, maybe this might be a bit more clear?:
``` suggestion
def __header_and_cpp_gen(param_info, cfg_name, header_file_pointer, cpp_file_pointer):
```

Reply:
Fixed